### PR TITLE
Add changelog for 0.17.0

### PR DESCRIPTION
**Pull request by straight-shoota**
Preview: https://github.com/crystal-lang/shards/blob/changelog/0.17.0/CHANGELOG.md

### CODE REVIEW

**Comment by straight-shoota** [comment]
Ecosystem test: https://github.com/crystal-lang/test-ecosystem/actions/runs/2024911221

**Comment by straight-shoota** [comment]
Fixed ecosystem test: https://github.com/crystal-lang/test-ecosystem/actions/runs/2025452567 
The remaining failing tests seem unrelated. The shards failures are because `fossil` is missing on the test system.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,21 @@
 # CHANGELOG
 
+## v0.17.0 - 2022-03-23
+
+- Add `make build` recipe ([#533](https://github.com/crystal-lang/shards/pull/533), thanks @straight-shoota)
+- Fix unexpected token compiler error match ([#532](https://github.com/crystal-lang/shards/pull/532), thanks @straight-shoota)
+- Honour `CRYSTAL` env var ([#534](https://github.com/crystal-lang/shards/pull/534), thanks @straight-shoota)
+- No longer depend of external git user config ([#536](https://github.com/crystal-lang/shards/pull/536), thanks @luislavena)
+- [CI] Update circleci xcode 13.2.1 ([#537](https://github.com/crystal-lang/shards/pull/537), thanks @straight-shoota)
+- Feat: output STDERR from the building process ([#540](https://github.com/crystal-lang/shards/pull/540), thanks @beta-ziliani)
```
Comment (Sija) [inline CHANGELOG.md:10]:
```suggestion
- Feat: output `STDERR` from the building process ([#540](https://github.com/crystal-lang/shards/pull/540), thanks @beta-ziliani)
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,21 @@
...
+- Fix grammar problems ([#543](https://github.com/crystal-lang/shards/pull/543), thanks @dinko-pehar)
+- Add fossil resolver ([#530](https://github.com/crystal-lang/shards/pull/530), thanks @MistressRemilia)
+- Add expanded local path to `shard.yml` error message in `PathResolver` ([#541](https://github.com/crystal-lang/shards/pull/541), thanks @straight-shoota)
+- Avoid user defined git template in resolver ([#528](https://github.com/crystal-lang/shards/pull/528), thanks @lzap)
+- Add run command ([#546](https://github.com/crystal-lang/shards/pull/546), thanks @luislavena)
+- re-enabled ~ support in path resolver ([#538](https://github.com/crystal-lang/shards/pull/538), thanks @masukomi)
```
Comment (Sija) [inline CHANGELOG.md:16]:
```suggestion
- Re-enabled `~` support in path resolver ([#538](https://github.com/crystal-lang/shards/pull/538), thanks @masukomi)
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,21 @@
...
+- Add `--jobs` (parallel git fetch) ([#539](https://github.com/crystal-lang/shards/pull/539), thanks @m-o-e)
```
Comment (Sija) [inline CHANGELOG.md:17]:
```suggestion
- Add `--jobs` flag (parallel git fetch) ([#539](https://github.com/crystal-lang/shards/pull/539), thanks @m-o-e)
```

File: man/shard.yml.5 [hunk 4]
```diff
@@ -223,7 +223,7 @@ authors:
 A restriction to indicate which are the supported crystal versions. This will
 usually express a lower and upper\-bound constraints (string, recommended)
 .sp
-When resolving dependencies, this information is not used. After dependencies
+When resolving dependencies, this information is not used. After dependecies
```
Comment (asterite) [inline man/shard.yml.5:226]:
```suggestion
When resolving dependencies, this information is not used. After dependencies
```

**Review by asterite — APPROVED** [review]
Looks good! A typo was introduced, so it should be reverted before merging.

**Pull request merged**